### ISSUE TITLE
[react/destructuring-assignment] False positives when using `render` function inside an object

### ISSUE THREAD

**Issue by qooban** [yannickcr/eslint-plugin-react#3022]
With `react/destructuring-assignment` rule turned on, we get false positives for the case when `render` function is defined inside an object. Seems like such function is treated the same way as a `render` function inside React class component and the first argument of such function is expected to be destructured before use.

Reproduction example (TypeScript):
```typescript
const test = {
    render(arg1: { id: string }, arg2: { id: string }) {
        return <div id={arg1.id} className={arg2.id} />;
    }
};
```
For such example ESLint reports: "Must use destructuring arg1 assignment(react/destructuring-assignment)". It does not report it for `arg2`.

`eslint-plugin-react` version: 7.24.0

**Comment by ljharb** [comment]
Indeed, an uncapitalized function shouldn’t be detected as a component, and render on a plain object also shouldn’t.

**Comment by ohhoney1** [comment]
#3001 fixed

**Comment by ljharb** [comment]
Fixed in #3001.
